File: ios/SimpleScout/Models/Place.swift
import Foundation

enum PlaceCategory: String, Codable, Equatable, Hashable {
    case hotel, restaurant, store, caterer, venue, pharmacy, medical, grocery, equipment, tech, transportation, other
}

enum PhotoTheme: String, Codable, Equatable, Hashable {
    case hotelLobby = "hotel-lobby"
    case hotelExterior = "hotel-exterior"
    case restaurantInterior = "restaurant-interior"
    case quickService = "quick-service"
    case catering
    case sportingGoods = "sporting-goods"
    case electronics
    case pharmacy
    case grocery
    case medical
    case venue
}

enum Hours: Codable, Equatable, Hashable {
    case open24h
    case closed
    case ranged(opens: String, closes: String)
}

/// Data that comes from a third-party provider (Google Places today). Never
/// the source of truth for anything proprietary — see docs/ARCHITECTURE.md §14.
struct PlaceExternalRef: Codable, Equatable {
    var provider: String = "google_places"
    let providerPlaceId: String
    var publicRating: Double? = nil
    var publicRatingCount: Int? = nil
    var priceLevel: Int? = nil // 0-4
    var phone: String? = nil
    var website: String? = nil
    var hoursToday: Hours? = nil
    let photoTheme: PhotoTheme
}

struct GroupSalesContact: Codable, Equatable {
    let name: String
    let role: String
    var phone: String? = nil
    var email: String? = nil
}

/// Proprietary, department-agnostic facts SimpleScout owns about a place.
struct AthleticsProfile: Codable, Equatable {
    var busParking: Bool? = nil
    var busParkingNotes: String? = nil
    var maxRecommendedGroup: Int? = nil
    var hasPrivateDining: Bool? = nil
    var hasMeetingSpace: Bool? = nil
    var earlyBreakfastAvailable: Bool? = nil
    var lateCheckoutAvailable: Bool? = nil
    var laundryAvailable: Bool? = nil
    var fitnessRoom: Bool? = nil
    var deliveryAvailable: Bool? = nil
    var pickupAvailable: Bool? = nil
    var cateringAvailable: Bool? = nil
    var onlineOrdering: Bool? = nil
    var dietaryOptions: [String] = []
    var groupSalesContact: GroupSalesContact? = nil
}

struct HotelDetails: Codable, Equatable {
    var totalRooms: Int? = nil
    var supportsRoomBlocks: Bool = true
}

struct RestaurantDetails: Codable, Equatable {
    let cuisine: String
    var familyStyle: Bool? = nil
    var buffet: Bool? = nil
    var reservationRequired: Bool? = nil
    var quickMealFriendly: Bool? = nil
    var estimatedPrepMinutes: Int? = nil
}

struct StoreDetails: Codable, Equatable {
    let storeType: String
    var likelyInventory: [String] = []
}

struct CatererDetails: Codable, Equatable {
    var minOrderCount: Int? = nil
    var leadTimeHours: Int? = nil
    var servesBreakfast: Bool? = nil
    var servesLunch: Bool? = nil
    var servesDinner: Bool? = nil
    var boxedMeals: Bool? = nil
    var buffet: Bool? = nil
    var estimatedCostPerPerson: Double? = nil
}

struct PlaceCapacity: Codable, Equatable {
    var maxPartySize: Int? = nil
    var idealPartySizeMin: Int? = nil
    var idealPartySizeMax: Int? = nil
}

struct Place: Identifiable, Codable, Equatable {
    let id: String
    let category: PlaceCategory
    let name: String
    let address: String
    let location: LatLng
    let universityId: String
    var external: PlaceExternalRef
    var athletics: AthleticsProfile
    var capacity: PlaceCapacity
    var hotel: HotelDetails? = nil
    var restaurant: RestaurantDetails? = nil
    var store: StoreDetails? = nil
    var caterer: CatererDetails? = nil
    var tags: [String] = []
}

File: src/models/place.ts
import type { LatLng } from "./university";

export type PlaceCategory =
  | "hotel"
  | "restaurant"
  | "store"
  | "caterer"
  | "venue"
  | "pharmacy"
  | "medical"
  | "grocery"
  | "equipment"
  | "tech"
  | "transportation"
  | "other";

/** Data that comes from a third-party provider (Google Places today). Never
 * the source of truth for anything proprietary — see docs/ARCHITECTURE.md §14. */
export interface PlaceExternalRef {
  provider: "google_places";
  providerPlaceId: string;
  publicRating?: number;
  publicRatingCount?: number;
  priceLevel?: 0 | 1 | 2 | 3 | 4;
  phone?: string;
  website?: string;
  hoursToday?: { opens: string; closes: string } | "closed" | "open_24h";
  photoTheme: PhotoTheme;
}

/** Deterministic placeholder art theme instead of hot-linked stock photography,
 * so the demo has no external asset dependency. */
export type PhotoTheme =
  | "hotel-lobby"
  | "hotel-exterior"
  | "restaurant-interior"
  | "quick-service"
  | "catering"
  | "sporting-goods"
  | "electronics"
  | "pharmacy"
  | "grocery"
  | "medical"
  | "venue";

/** Proprietary, department-agnostic facts SimpleScout owns about a place.
 * See docs/ARCHITECTURE.md §14. */
export interface AthleticsProfile {
  busParking?: boolean;
  busParkingNotes?: string;
  maxRecommendedGroup?: number;
  hasPrivateDining?: boolean;
  hasMeetingSpace?: boolean;
  earlyBreakfastAvailable?: boolean;
  lateCheckoutAvailable?: boolean;
  laundryAvailable?: boolean;
  fitnessRoom?: boolean;
  deliveryAvailable?: boolean;
  pickupAvailable?: boolean;
  cateringAvailable?: boolean;
  onlineOrdering?: boolean;
  dietaryOptions?: string[];
  groupSalesContact?: { name: string; role: string; phone?: string; email?: string };
}

export interface HotelDetails {
  totalRooms?: number;
  supportsRoomBlocks: boolean;
}

export interface RestaurantDetails {
  cuisine: string;
  familyStyle?: boolean;
  buffet?: boolean;
  reservationRequired?: boolean;
  quickMealFriendly?: boolean;
  estimatedPrepMinutes?: number;
}

export interface StoreDetails {
  storeType: "sporting_goods" | "electronics" | "pharmacy" | "grocery" | "hardware" | "printing";
  likelyInventory?: string[];
}

export interface CatererDetails {
  minOrderCount?: number;
  leadTimeHours?: number;
  servesBreakfast?: boolean;
  servesLunch?: boolean;
  servesDinner?: boolean;
  boxedMeals?: boolean;
  buffet?: boolean;
  estimatedCostPerPerson?: number;
}

export interface PlaceCapacity {
  maxPartySize?: number;
  idealPartySizeMin?: number;
  idealPartySizeMax?: number;
}

export interface Place {
  id: string;
  category: PlaceCategory;
  name: string;
  address: string;
  location: LatLng;
  universityId: string;
  external: PlaceExternalRef;
  athletics: AthleticsProfile;
  capacity: PlaceCapacity;
  hotel?: HotelDetails;
  restaurant?: RestaurantDetails;
  store?: StoreDetails;
  caterer?: CatererDetails;
  tags: string[];
}
